### PR TITLE
Add medical_tests module to donor_organism. Fixes #1562

### LINKED ISSUE #1562: Add medical_tests module in donor_organism
**For which schema is a change/update being suggested?**

I would like to request an update to the `donor_organism` schema.

**What should the change/update be?**

Add a new module called `medical_tests` that would record results & parameters of specific medical test of interest for the bionetworks, that will be part of their proposed Tier 2 fields. For start, we have the list of the Lung bionetwork proposed test-results.

This update constitutes a **major** change to the schema(s) it affects.

**What new field(s) need to be changed/added?**

The module with include the following list of fields:
* PFT_method
* PFT_age
* PFT_time_point
* PFT_relative_time_point
* FEV1_predicted
* FEV1_preBD
* FEV1_postBD
* FEV1_preBD_predicted_percent
* FEV1_postBD_predicted_percent
* FVC_predicted
* FVC_preBD
* FVC_postBD
* FVC_preBD_predicted_percent
* FVC_postBD_predicted_percent
* FEV1_FVC_ratio_preBD
* FEV1_FVC_ratio_postBD
* FRC_abs
* FRC_predicted_percent
* RV
* RV_predicted_percent
* IC
* IC_predicted_percent
* DLCO
* DLCO_predicted_percent
* KCO
* KCO_predicted_percent
<details><summary>extended list</summary>


| **Test field**                       | **Description**                                                                                                                                                                                                                                               | **Type** | **Mandatory** | **Allowed Values**                                      | **Example**                           |
|-------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------|---------------|----------------------------------------------------------|---------------------------------------|
| PFT_method                          | Indicate which Pulmonary Function Testings (PFTs) methods were performed. Please indicate all tests performed.                                                                                                                   | enum     | no            | spirometry; plethysmography; DLCO_KCO_test               | spirometry; plethysmography; DLCO_KCO_test |
| PFT_age                             | Indicate the age of the subject at the time point of pulmonary function parameter assessment in years, if known.                                                                                                                                              | numeric  | no            | 10;34;68                                                  | 10;34;68                              |
| PFT_time_point                      | Indicate at which date the pulmonary function parameters were recorded as YYYY-MM-DD if available. If this information is not shareable, please indicate "not_shareable".                                                                                     | string   | no            |                                                          |                                       |
| PFT_relative_time_point             | If timepoint of sample collection (sample_collection_time_point) and/or timepoint of PFT recording (PFT_time_point) is not available, indicate how many days prior or after sample collection the pulmonary function parameters were assessed if available. | numeric  | no            |                                                          |                                       |
| FEV1_predicted                      | Indicate the predicted Forced Expiratory Volume in 1 second (FEV1) in milliliters, based on patient age, sex, height, and ethnicity, if available.                                                                                                            | numeric  | no            |                                                          |                                       |
| FEV1_preBD                          | Indicate the measured FEV1 in milliliters before the administration of a bronchodilator if available.                                                                                                                                                        | numeric  | no            |                                                          |                                       |
| FEV1_postBD                         | Indicate the measured FEV1 in milliliters after the administration of a bronchodilator if available.                                                                                                                                                         | numeric  | no            |                                                          |                                       |
| FEV1_preBD_predicted_percent        | Indicate the percentage of the predicted FEV1 achieved before bronchodilator administration if available.                                                                                                                                                     | numeric  | no            |                                                          |                                       |
| FEV1_postBD_predicted_percent       | Indicate the percentage of the predicted FEV1 achieved after bronchodilator administration if available.                                                                                                                                                      | numeric  | no            |                                                          |                                       |
| FVC_predicted                       | Indicate the predicted Forced Vital Capacity (FVC) in milliliters, based on patient age, sex, height, and ethnicity, if available.                                                                                                                            | numeric  | no            |                                                          |                                       |
| FVC_preBD                           | Indicate the measured FVC in milliliters before bronchodilator administration if available.                                                                                                                                                                  | numeric  | no            |                                                          |                                       |
| FVC_postBD                          | Indicate the measured FVC in milliliters after bronchodilator administration if available.                                                                                                                                                                   | numeric  | no            |                                                          |                                       |
| FVC_preBD_predicted_percent         | Indicate the percentage of the predicted FVC achieved before bronchodilator administration if available.                                                                                                                                                      | numeric  | no            |                                                          |                                       |
| FVC_postBD_predicted_percent        | Indicate the percentage of the predicted FVC achieved after bronchodilator administration if available.                                                                                                                                                       | numeric  | no            |                                                          |                                       |
| FEV1_FVC_ratio_preBD                | Indicate the ratio of FEV1 to FVC before bronchodilator administration if available.                                                                                                                                                                         | numeric  | no            |                                                          |                                       |
| FEV1_FVC_ratio_postBD               | Indicate the ratio of FEV1 to FVC after bronchodilator administration if available.                                                                                                                                                                          | numeric  | no            |                                                          |                                       |
| FRC_abs                             | Indicate the Functional Residual Capacity (FRC) in absolute milliliters if available.                                                                                                                                                                        | numeric  | no            |                                                          |                                       |
| FRC_predicted_percent               | Indicate the percentage of the predicted FRC, based on patient demographics, if available.                                                                                                                                                                    | numeric  | no            |                                                          |                                       |
| RV                                  | Indicate the Residual Volume (RV) in milliliters if available.                                                                                                                                                                                               | numeric  | no            |                                                          |                                       |
| RV_predicted_percent                | Indicate the percentage of the predicted RV, based on patient demographics, if available.                                                                                                                                                                     | numeric  | no            |                                                          |                                       |
| IC                                  | Indicate the Inspiratory Capacity (IC) in milliliters if available.                                                                                                                                                                                          | numeric  | no            |                                                          |                                       |
| IC_predicted_percent                | Indicate the percentage of the predicted IC, based on patient demographics, if available.                                                                                                                                                                     | numeric  | no            |                                                          |                                       |
| DLCO                                | Indicate the Diffusing Capacity of the Lung for Carbon Monoxide (DLCO) in milliliters if available.                                                                                                                                                          | numeric  | no            |                                                          |                                       |
| DLCO_predicted_percent              | Indicate the percentage of the predicted DLCO, based on patient demographics, if available.                                                                                                                                                                   | numeric  | no            |                                                          |                                       |
| KCO                                 | Indicate the transfer coefficient of the lung for carbon monoxide (KCO) in mmol/min/kPa/L if available.                                                                                                                                                       | numeric  | no            |                                                          |                                       |
| KCO_predicted_percent               | Indicate the percentage of the predicted KCO, based on patient demographics, if available.                                                                                                                                                                    | numeric  | no            |                                                          |                                       |

</details>

**Why is the change requested?**

Contributor should be able to collect data about Pulmonary Function Tests, in a standardised and comparable way. Along with test results, test metadata will be recorded as well (PFT_*)

Tier 2 metadata will include biological metadata fields, specific for each bionetwork.
We decided that instead of introducing a bionetwork specific module, we prefer to divide the Tier 2 fields in content specific modules, like "medical_tests", "disease_state" or "treatments" to take advantage of potentially overlaps between bionetworks. From conversations we already have had with different bionetworks, we expect to have multiple test result fields requested. 
Therefore, here we introduce the `medical_tests` module.

Since we expect a large number of `medical_test` fields from other bionetworks as well, we decided to have them in a separate module. Also, we could add it in the `medical_history` module, however then we would have a module in a module structure, which is not encouraged in our schema.

We already have a free-text field for `test_results` in the `medical_history` module, however it is a free-text field with no standard way to define the test, units and test value of the test. We will not remove it as it is impossible to record all the available test results in the new module, but it will not be offered for bionetwork contributors who have test options in the new module.

The definitions and controlled vocabulary are provided by the Lung bionetwork team.

### PR DESCRIPTION
**Pull request by arschat**
### Release notes

For `donor_organism` schema, add a `medical_tests` module that includes the following fields.
* pft_method
* pft_age
* pft_time_point
* pft_relative_time_point
* fev1_predicted
* fev1_prebd
* fev1_postbd
* fev1_prebd_predicted_percent
* fev1_postbd_predicted_percent
* fvc_predicted
* fvc_prebd
* fvc_postbd
* fvc_prebd_predicted_percent
* fvc_postbd_predicted_percent
* fev1_fvc_ratio_prebd
* fev1_fvc_ratio_postbd
* frc_abs
* frc_predicted_percent
* rv
* rv_predicted_percent
* ic
* ic_predicted_percent
* dlco
* dlco_predicted_percent
* kco
* kco_predicted_percent

Extended list with descriptions and options in #1562 

### Reviews requested

- Need 5 Reviewers to approve because this is a **major** update

### CODE REVIEW

File: json_schema/module/biomaterial/medical_tests.json [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+            "maximum": 150,
+            "minimum": 0,
+            "$comments": "pulmonary_function_parameter; lung"
+        },
+        "PFT_time_point": {
+            "user_friendly": "Date at Pulmonary Function Test",
+            "description": "Indicate at which date the pulmonary function parameters were recorded as DD/MM/YYYY if available. If this information is not shareable, please indicate 'not_shareable'.",
+            "type": "string",
+            "example": "14/11/2024; 31/01/2019; not_shareable",
+            "pattern": "^((0[1-9]|[12][0-9]|3[01])[- \\.\\/](0[1-9]|1[0-2])[- \\.\\/](19|20)\\d\\d|not_applicable)$",
```
[the remark this answers is not in the document]
Comment (idazucchi) [reply json_schema/module/biomaterial/medical_tests.json:45]: the format for the date here is different from all the other dates in the schema, do we want to swap it?

**Review by ncalvanese1 — APPROVED** [review]

**Comment by hannes-ucsc** [comment]
I'll hold off until EBI folks approve.

File: json_schema/module/biomaterial/medical_tests.json [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+        "DLCO_predicted_percent": {
+            "user_friendly": "DLCO percent of predicted",
+            "description": "Indicate the percentage of the predicted DLCO, based on patient demographics, if available.",
+            "type": "number",
+            "example": "88; 95; 82",
+            "maximum": 200,
+            "minimum": 0,
+            "$comments": "pulmonary_function_parameter; lung"
+        },
+        "KCO": {
```
Comment (ESapenaVentura) [inline json_schema/module/biomaterial/medical_tests.json:240]: please use snake_case formatting for the field name (e.g. `kco`, `kco_predicted_percent`). I know for some gene symbols/acronyms can be a bit weird, but we use this convention and the proper capitalization can be used in user_friendly.

**Review by ESapenaVentura — CHANGES_REQUESTED** [review]
I have some comments about the naming of the fields, some missing values in the update log (And versions.json) and the usage of the `$comment` keyword

**Review by amnonkhen — APPROVED** [review]

**Comment by arschat** [comment]
Attribute names should convey intent, so we changed the `$comment` field to `bionetwork`.
Most of these Tier 2 fields are bionetwork specific and may not be relevant for other bionetworks. The intent is to record the bionetwork that requested this field, so we could generate bionetwork tailored spreadsheets directly from the schema.

**Review by amnonkhen — APPROVED** [review]

**Review by hannes-ucsc — APPROVED** [review]

**Review by NoopDog — APPROVED** [review]

**Pull request merged**